src/scene/layer.js

```diff
@@ -400,15 +400,20 @@ Layer.prototype.addMeshInstances = function (meshInstances, skipShadowCasters) {
     var casters = this.shadowCasters;
     for (var i = 0; i < meshInstances.length; i++) {
         m = meshInstances[i];
+
         mat = m.material;
         if (mat.blendType === BLEND_NONE) {
             arr = this.opaqueMeshInstances;
         } else {
             arr = this.transparentMeshInstances;
         }
 
+        // test for meshInstance in both arrays, as material's alpha could have changed since LayerComposition's update to avoid duplicates
         // TODO - following uses of indexOf are expensive, to add 5000 meshInstances costs about 70ms on Mac. Consider using Set.
```

Comment:
As we are doing two indexofs, should we consider this TODO as we've now added to the cost?

Reply:
that's separate PR, I don't want to get to it now.